api/src/adapters/aws/sesv2_adapter.py

```diff
@@ -18,45 +23,93 @@ class SESV2Response(BaseModel):
     suppressed_destination_summaries: list[SuppressedDestination] = Field(
         alias="SuppressedDestinationSummaries", default_factory=list
     )
+    next_token: str | None = Field(alias="NextToken", default=None)
 
 
 class BaseSESV2Client(ABC, metaclass=ABCMeta):
     @abstractmethod
-    def list_suppressed_destinations(self) -> SESV2Response:
+    def list_suppressed_destinations(self, start_time: datetime | None = None) -> SESV2Response:
         pass
 
 
 class SESV2Client(BaseSESV2Client):
     def __init__(self) -> None:
-        self.client = boto3.client("sesv2")
+        self.client = get_boto_sesv2_client()
+
+    def list_suppressed_destinations(self, start_time: datetime | None = None) -> SESV2Response:
+        request_params: dict[str, datetime | str] = {}
+        if start_time:
+            request_params["StartDate"] = start_time
+
+        all_summaries = []
+        next_token = None
+
+        try:
+            while True:
```

Comment:
As a safety net, could we put some way for this to eventually break out after like 100 iterations?

Just something like:
```py
iterations = 0
while True:
     iterations += 1

     # existing logic

     if iterations > 100:
         logger.error("Stopping iteration of response from list suppression API after 100 iterations")
         break
```
It's unlikely, but the alternative is the code endlessly calling if there were some edge case we weren't aware of.